clients/src/test/java/org/apache/kafka/common/metrics/OtlpMetricsReporter.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.metrics;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.opentelemetry.proto.common.v1.AnyValue;
+import io.opentelemetry.proto.common.v1.KeyValue;
+import io.opentelemetry.proto.metrics.v1.ResourceMetrics;
+import io.opentelemetry.proto.metrics.v1.MetricsData;
+import io.opentelemetry.proto.resource.v1.Resource;
+
+import org.apache.kafka.server.authorizer.AuthorizableRequestContext;
+import org.apache.kafka.common.requests.RequestContext;
+import org.apache.kafka.clients.ClientTelemetryUtils;
+import org.slf4j.LoggerFactory;
+import org.slf4j.Logger;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+
+/**
+ * MetricsReporter that aggregates Opentelemetry Protocol(OTLP) metrics, enhance them with additional client labels
+ * and forwards them via gRPC Client to an external OTLP receiver.
+ */
+public class OtlpMetricsReporter implements MetricsReporter {
+
+    private static final Logger log = LoggerFactory.getLogger(OtlpMetricsReporter.class);
+
+    private MetricsGrpcClient grpcService;
+
+    public boolean appendClientLabels = true;
+
+    public static final String APPEND_CLIENT_LABELS = "APPEND_CLIENT_LABELS";
```

Comment:
This constant name does not provide much explanatory value over the raw string.  Can you change the variable name to describe the meaning of this field, else just in-line the raw string.

i.e.
```
int INCHES_PER_FOOT = 12
```
vs.
```
int TWELVE = 12
```